### PR TITLE
UnZip Remove files based on exclusion list (#3872)

### PR DESCRIPTION
**Pull request by rootsradics**

### CODE REVIEW

File: src/pyload/plugins/extractors/UnZip.py [hunk 1]
```diff
@@ -80,7 +81,11 @@ def extract(self, password=None):
         try:
             with zipfile.ZipFile(self.filename, "r") as z:
                 z.setpassword(password)
-                z.extractall(self.dest)
+                members = (member for member in z.namelist()  
+                           if not any(re.compile(fnmatch.translate(exclusion)).match(member)
+                           for exclusion in self.excludefiles)
```
Comment (GammaC0de) [inline src/pyload/plugins/extractors/UnZip.py:86]: Why not just:
`if not any(fnmatch.fnmatch(member, exclusion)`
?

File: src/pyload/plugins/extractors/UnZip.py [hunk 2]
```diff
@@ -80,7 +81,11 @@ def extract(self, password=None):
...
+                          )
+                z.extractall(self.dest,members=members)
                 self.files = [os.path.join(self.dest, _f)
                               for _f in z.namelist()
                               if _f[-1] != os.path.sep]
```
Comment (GammaC0de) [inline src/pyload/plugins/extractors/UnZip.py:91]: Should be changed to:
`if _f[-1] != os.path.sep and _f in members]`

**Comment by GammaC0de** [comment]
Also, plugin version should be increased.

**Comment by rootsradics** [comment]
Should be OK too !
Sorry, I'm quite new to Python, and I'm missing some API knowledge :)
I will try to pull request with the fix you proposed ( the one below,
and the two others (bump version and add members to the returned files list)

Le 07/01/2021 à 23:08, GammaC0de a écrit :

**Pull request closed without merging**